lib/cloudregister/registerutils.py

```diff
@@ -173,6 +175,96 @@ def clear_rmt_as_scc_proxy_flag():
         pass
 
 
+# ----------------------------------------------------------------------------
+def clean_payg_extensions():
+    """Remove non free extensions files from /etc/products.d/."""
+    extensions = get_extensions()
+    products_paths = glob.glob('/etc/products.d/*.prod')
```

Comment:
We should let zypper tell us what products are on the system. zypper already has a reliable way to do this and verifies things along the way. If changes are made to this type of identification there will be a feature ticket that will get implemented in zypper but we will be forgotten and our code will break. We should depend on teh distribution implementation in this case.

And we already have an implementation for this ;) `get_installed_products()` in registerutils.

Reply:
and as I have learned SUSEConnect installs and de-installs packages the correct way to de-register would then be another SUSEConnect call

```
SUSEConnect --de-register --product SLES-LTSS/15.4/x86_64
```

Reply:
Yes, that is the approach I took